### PR TITLE
inter-partition pull binlog

### PR DESCRIPTION
**Pull request by chaychen2005**

### CODE REVIEW

File: libledger/DBInitializer.cpp [hunk 1]
```diff
@@ -142,53 +142,59 @@ void DBInitializer::initLevelDBStorage()
...
+    int64_t startNum = -1;
+    auto memoryTableFactory = m_tableFactoryFactory->newTableFactory(dev::h256(), startNum);
     Table::Ptr tb = memoryTableFactory->openTable(SYS_CURRENT_STATE, false);
     auto entries = tb->select(SYS_KEY_CURRENT_NUMBER, tb->newCondition());
     if (entries->size() > 0)
     {
         auto entry = entries->get(0);
         std::string currentNumber = entry->getField(SYS_VALUE);
-        num = boost::lexical_cast<int64_t>(currentNumber.c_str());
+        startNum = boost::lexical_cast<int64_t>(currentNumber.c_str());
```
Comment (bxq2011hust) [inline libledger/DBInitializer.cpp]:
```suggestion
        startNum = boost::lexical_cast<int64_t>(currentNumber);
```

Comment (bxq2011hust) [reply libledger/DBInitializer.cpp]: help

File: libledger/DBInitializer.cpp [hunk 2]
```diff
@@ -142,53 +142,59 @@ void DBInitializer::initLevelDBStorage()
...
-    // getMissingBlocksFromBinLog return (num,latest]
-    auto blocksData = _binaryLogger->getMissingBlocksFromBinLog(num);
-    DBInitializer_LOG(INFO) << LOG_DESC("recover from binary logs") << LOG_KV("blockNumber", num);
-
-    if (blocksData->size() > 0)
+    // getMissingBlocksFromBinLog from (startNum,lastBlockNum]
+    int64_t lastBlockNum = _binaryLogger->getLastBlockNum();
+    DBInitializer_LOG(INFO) << LOG_DESC("recover from binary logs") << LOG_KV("startNum", startNum)
+                            << LOG_KV("endNum", lastBlockNum);
+    int64_t interval = 10;
```
Comment (bxq2011hust) [inline libledger/DBInitializer.cpp]: what about 100, to decrease disk IO ?

File: libledger/DBInitializer.cpp (hunk 1 above)
Comment (chaychen2005) [reply libledger/DBInitializer.cpp]: Done.

**Pull request merged**